components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -18,77 +25,62 @@ import mozilla.components.feature.toolbar.internal.URLRenderer
 @Suppress("TooManyFunctions")
 class ToolbarPresenter(
     private val toolbar: Toolbar,
-    private val sessionManager: SessionManager,
-    private val sessionId: String? = null,
+    private val store: BrowserStore,
+    private val customTabId: String? = null,
     urlRenderConfiguration: ToolbarFeature.UrlRenderConfiguration? = null
-) : SelectionAwareSessionObserver(sessionManager) {
+) {
 
     @VisibleForTesting
     internal var renderer = URLRenderer(toolbar, urlRenderConfiguration)
 
+    private val scope = MainScope()
+
     /**
      * Start presenter: Display data in toolbar.
      */
     fun start() {
-        observeIdOrSelected(sessionId)
-        initializeView()
-
         renderer.start()
-    }
 
-    override fun stop() {
-        super.stop()
-
-        renderer.stop()
+        scope.launch {
+            store.channel()
+                .map { state -> state.findCustomTabOrSelectedTab(customTabId) }
+                .consumeEachIfChanged(
+                    then = { render(it) },
+                    otherwise = { clear() }
```

Comment:
I think `otherwise` doesn't read right here. It sounds like `if changed to do this and otherwise that`, but it's really covering the null case, right? Should we rename this to `onNull` or `ifNull`, or add a separate method for handling the null case?